Wrap Player.prototype.options

gkatsev opened the pull request:
This is necessary because some users relied on the behavior of
options().children to be an object that returned the associated object
but in 5.0 it is now an array for the default components, such as the
player. So, we want to wrap the function to attach the properies of the
children to the children array so both would be available.
This is a copy of https://github.com/forbesjo/videojs-4-5-compatibility/pull/2

gkatsev committed:
Wrap Player.prototype.options
This is necessary because some users relied on the behavior of
options().children to be an object that returned the associated object
but in 5.0 it is now an array for the default components, such as the
player. So, we want to wrap the function to attach the properies of the
children to the children array so both would be available.

misteroneill commented on src/plugin.js:
```diff
@@ -41,6 +41,19 @@
+    if (Object.prototype.toString.call(options.children) === '[object Array]') {
+      for (var i = 0; i < options.children.length; i++) {
+        var childName = options.children[i];
+        options.children[childName] = this.getChild(childName);
```
Is this supposed to be exposing the child object or the options for the child object?

gkatsev replied:
Hm... yeah, we probably want to grab the options of the component instead of the component itself.

gkatsev replied:
Fixed.

gkatsev committed:
Grab options_ and not the component itself
We want to use options_ instead of options() because otherwise we'll get
a lot of deprecation warnings for each child. This way, we get the
options but only one deprecation warning for calling player.options()
in general.

misteroneill commented:
LGTM
